src/Utilities.jl

```diff
@@ -84,19 +84,7 @@ end
 
 ## non-allocating but code duplication
 
-Base.@propagate_inbounds function p_v_idx(p::Tuple{T1,T2,T3}, i) where {T1<:AbstractArray,T2,T3}
-    @view p[1][:, i]
-end
-
-Base.@propagate_inbounds function p_v_idx(p::Tuple{T1,T2,T3}, i) where {T1<:AbstractVector{T4} where {T4},T3,T2}
-    p[1][i]
-end
-
-Base.@propagate_inbounds function p_v_idx(p::Tuple{T1,T2,T3}, i) where {T1,T2,T3}
-    p[1]
-end
-
-Base.@propagate_inbounds function p_v_idx(p::Tuple{T1,T2}, i) where {T1<:AbstractArray,T2}
+@inline Base.@propagate_inbounds function p_v_idx(p::Tuple{T1,T2}, i) where {T1<:AbstractArray,T2}
```

Comment:
```suggestion
Base.@propagate_inbounds function p_v_idx(p::Tuple{T1,T2}, i) where {T1<:AbstractArray,T2}
```
inbound propagation requires inlined functions, therefore `@propagate_inbounds` is allready the stronger hint towards inlining